### DISCUSSION TITLE
Possible bug: Indirect Go modules seem to be enabled as default in Renovate self-hosted

### DISCUSSION THREAD

**Discussion by mikaello** [renovatebot/renovate#20174] [Suggest an Idea]
### Type of discussion.

I just want to chat

### Tell us more.

We have just upgraded our self hosted Renovate to [v3.0.0](https://github.com/mend/renovate-on-prem/releases/tag/3.0.0). We are using it on a self hosted GitLab Enterprise Edition 15.7.6-ee.

We got a large amount of updates for indirect Go dependencies after the upgrade to `v3.0.0`, which seems to have been added in [Renovate 34.58.0](https://github.com/renovatebot/renovate/releases/tag/34.58.0). The documentation added in [the corresponding PR](https://github.com/renovatebot/renovate/pull/19431) says:

> Indirect updates are disabled by default. To enable them, add a package rule such as:
> 
> ```json
> {
>   "packageRules": [
>     {
>       "matchManagers": ["gomod"],
>       "matchDepTypes": ["indirect"],
>       "enabled": true
>     }
>   ]
> }
> ```

A config we have not added. But if I add:

```json
{
  "packageRules": [
    {
      "matchManagers": ["gomod"],
      "matchDepTypes": ["indirect"],
      "enabled": false
    }
  ]
}
```

... Renovate stopped creating indirect dependency updates, and already created pull requests were closed.

<details><summary>Click to see log of an actual update creating a pull request</summary>

```json
{
  "name": "renovate",
  "hostname": "renovate-5bdfdb5cd9-4smsp",
  "pid": 8,
  "level": 20,
  "logContext": "i0mcVAyXo",
  "repository": "my-go-repository",
  "baseBranch": "main",
  "config": {
    "gomod": [
      {
        "packageFile": "go.mod",
        "deps": [
          {
            "managerData": {
              "lineNumber": 38,
              "multiLine": true
            },
            "depName": "github.com/go-stack/stack",
            "depType": "indirect",
            "currentValue": "v1.8.0",
            "datasource": "go",
            "enabled": false,
            "depIndex": 34,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "v1.8.1",
                "newValue": "v1.8.1",
                "releaseTimestamp": "2021-08-18T18:48:21.000Z",
                "newMajor": 1,
                "newMinor": 8,
                "updateType": "patch",
                "branchName": "renovate-github.com-go-stack-stack-1.x"
              }
            ],
            "warnings": [],
            "versioning": "semver",
            "sourceUrl": "https://github.com/go-stack/stack",
            "currentVersion": "v1.8.0",
            "isSingleVersion": true,
            "fixedVersion": "v1.8.0"
          }
        ]
      }
    ]
  },
  "msg": "packageFiles with updates",
  "time": "2023-02-02T12:04:58.850Z",
  "v": 0
}
```

</details>

In renovate config I have previously specified `{ "matchManagers": ["gomod"], "enabled": true },`, but I don't think this should enable indirect dependencies?

**Comment by rarkins** [comment]
Yes, that rule in your config enables all updates, including transitive

**Reply by mikaello** [reply]
I see. Thanks for your swift reply.

For future reference:

In addition to 

```json
      {
          "matchManagers": ["gomod"],
          "enabled": true
      },
```

.. does also

```json
      {
          "matchDatasources": ["go"],
          "enabled": true
      },
```

.. seem to enable indirect updates.
